pkg/resource/deploy/deployment_executor.go

```diff
@@ -188,7 +188,6 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context) (*Plan, error)
 	if err != nil {
 		return nil, err
 	}
-	defer src.Close()
```

Comment:
`src.Close()` collects errors due to program exit. This defer call had two issues;

1. we did not collect the error and propagate it
2. even if we did do that here, it's too late, we're finished writing out events and we want the error to go into the event log.

Instead we now close the source when it gets the signal on `finChan` https://github.com/pulumi/pulumi/pull/19837/commits/3e8d7212ed0187d40b4c7f27f1f3cb8c5c812d00